src/modules/m_sqloper.cpp

```diff
@@ -121,40 +144,47 @@ class ModuleSQLOper : public Module
 		else
 			SQL.SetProvider("SQL/" + dbid);
 
-		hashtype = tag->getString("hash");
-		query = tag->getString("query", "SELECT hostname as host, type FROM ircd_opers WHERE username='$username' AND password='$password' AND active=1;");
+		query = tag->getString("query", "SELECT * FROM ircd_opers WHERE active=1;");
+		const std::string noexist = "null"; //Because OpGetQuery never returns something to ModuleSQLOper, it is resposible for executing /OPER if necessary. Therefore, it requires params such as username and pwd that aren't necessary if it's updating for /REHASH.
+
+		GetOLines(false, noexist, noexist, noexist); //Update sqloper list from the database.
+		return;
+	}
+
+	CullResult cull()
+	{
+		ServerInstance->Config->oper_blocks = conf_blocks; //Reset oper blocks to those from the config
+		return Module::cull();
 	}
 
 	ModResult OnPreCommand(std::string &command, std::vector<std::string> &parameters, LocalUser *user, bool validated, const std::string &original_line) CXX11_OVERRIDE
 	{
-		if (validated && command == "OPER" && parameters.size() >= 2)
+		if (validated && command == "OPER" && parameters.size() >= 2 && original_line != "SQLOPER") //The line will be SQLOPER only if this module already reloaded the DB and is recalling /oper for other hooks
```

Comment:
Rather than checking original_line here can't you have a bool `active` sentinel in the class? The core (which calls events) is single threaded so this will work fine unless you need to modify it from the SQL thread.